src/core_modules/capture-core/components/D2Form/field/validators/getValidators.js

```diff
@@ -115,7 +115,7 @@ const validatorsForTypes = {
         {
             validator: isValidDate,
             message: errorMessages.DATE,
-            type: validatorTypes.TYPE_BASE,
+            type: validatorTypes.INTERNAL,
```

Comment:
Every error expect future date should have validatorTypes.TYPE_BASE. Future date should have TYPE_EXTENDED. Probably easiest and most readable if we make two validators so we don't have to implement new logic here for determining the type (we have two validators now, but the second one about the allowFutureDate is not implemented the way we want it)

The reason is that we depending on the program configuration, can accept saving an incomplete form. But the BASE-validators must always pass (i.e., your date field should be a date, your number field should be a number etc.)